cachi2/core/package_managers/gomod.py

```diff
@@ -548,10 +551,11 @@ def fetch_gomod_source(request: Request) -> RequestOutput:
         )
 
     env_vars = {
-        "GOCACHE": {"value": "deps/gomod", "kind": "path"},
-        "GOPATH": {"value": "deps/gomod", "kind": "path"},
-        "GOMODCACHE": {"value": "deps/gomod/pkg/mod", "kind": "path"},
-        "GOTOOLCHAIN": {"value": "local", "kind": "literal"},
+        "GOCACHE": "${output_dir}/deps/gomod",
+        "GOPATH": "${output_dir}/deps/gomod",
+        "GOMODCACHE": "${output_dir}/deps/gomod/pkg/mod",
+        "GOPROXY": "file://${GOMODCACHE}/cache/download",
+        "GOTOOLCHAIN": "local",
```

Comment:
Thank you for spotting this, you're right. When I did this I focused solely on enforcing a consistent "base" behaviour of the target image which can default either to `GOTOOLCHAIN=local` or `GOTOOLCHAIN=auto` depending on how it was packaged on that distro and I didn't like that because it can lead to negatively surprising results. However, I then neglected the technical aspect of how the hermetic builds work and falsely assumed the user would override our value to their liking just fine, but it is the other way around actually, whatever users configure will be overriden once they do a "source" on our env file. I'll drop this.